src/v2/upward.rs

```diff
@@ -242,8 +242,8 @@ where
 }
 
 #[derive(Debug, thiserror::Error)]
-#[error("Encountered unknown data from the Node API, which is required to be known.")]
-pub struct UnknownDataError;
+#[error("Encountered unknown data from the Node API on type `Upward::<{0}>::Unknown`, which is required to be known. Upgrade `concordium_rust_sdk` version.")]
+pub struct UnknownDataError(pub String);
```

Comment:
I think we can get away with using static str here, since it must be known at compile time
```suggestion
pub struct UnknownDataError(&'static str);
```